libkbfs/interfaces.go

```diff
@@ -1518,11 +1515,15 @@ type BareRootMetadata interface {
 	GetTLFCryptKeyParams(keyGen KeyGen, user keybase1.UID, key CryptPublicKey) (
 		TLFEphemeralPublicKey, EncryptedTLFCryptKeyClientHalf,
 		TLFCryptKeyServerHalfID, bool, error)
-	// IsValidAndSigned verifies the BareRootMetadata given the current
-	// user and device (identified by the KID of the device verifying
-	// key), checks the writer signature, and returns an error if a
-	// problem was found.
-	IsValidAndSigned(codec Codec, crypto cryptoPure,
+	// IsValidAndSigned verifies the BareRootMetadata, checks the
+	// writer signature, and returns an error if a problem was
+	// found.
+	IsValidAndSigned(codec Codec, crypto cryptoPure) error
+	// IsLastModifiedBy verifies that the BareRootMetadata is written by
+	// the given user and device (identified by the KID of the device
+	// verifying key), and returns an error if not. Should be called only
```

Comment:
There aren't any side effects, but I was trying to express that IsValidAndSigned should be the first thing checked on an RMD from an untrusted source, before doing anything else with it, and that the verifying keys should be validated somehow, either by using IsLastModifiedBy or checking against the KBPKI. Modified comments accordingly.